tfhe/src/integer/server_key/radix_parallel/vector_find.rs

```diff
@@ -96,7 +96,7 @@ impl ServerKey {
             max_output_value.ceil_ilog2()
         };
         let num_blocks_to_represent_values =
-            num_bits_to_represent_output_value.div_ceil(num_bits_in_message);
+            num_bits_to_represent_output_value.div_ceil(num_bits_in_message) + 1;
```

Comment:
I don't understand this fix
Maybe add a comment to explainthis `+1`

Reply:
Actually the +1 should be on the ceil_ilog2 : `ceil_ilog2()` replaced by `ilog2() + 1`

The get the number of bits that allows representing a number  we need the `log2 + 1` or `ceil_ilog2 + u32::from(value.is_power_of_two)`:

with ilog2 + 1:
- value 3 (0b11) -> ilog2 =  1, + 1 = 2 => 2 bits to represent the value
- value 4 (0b100) -> ilog2 =  2, + 1 = 3 => 3 bits to represent the value

with ceil_ilog2 + u32::from(value.is_power_of_two):
- value 3 (0b11) -> ceil_ilog2 =  2, + 0 = 2 => 2 bits to represent the value
- value 4 (0b100) -> ceil_ilog2 =  2, + 1 = 3 => 2 bits to represent the value

Reply:
Or `(max_output_value+1).ceil_ilog2()` which IMO is more clear (and equivalent I believe)

Reply:
with `(max_output_value+1).ceil_ilog2()` we would have to handle the overflow case when max_output == CLEAR::MAX

Reply:
unless we change
```rust
let num_bits_to_represent_output_value = if max_output_value == Clear::ZERO {
    1
} else {
    max_output_value.ilog2() + 1
};
```
to 

```rust
let (max_output_value_plus_1, did_overflow) = max_output_value.overflowing_add(1);
let num_bits_to_represent_output_value = if did overflow {
      Clear::BITS
  } else {
      max_output_value_plus_1.ceil_ilog2()
  };
```

Reply:
Or 
```
let num_bits_to_represent_output_value = if max_output_value == Clear::MAX {
    Clear::BITS
} else {
   (max_output_value+1).ceil_ilog2()
};
```